src/filterFlags.js

```diff
@@ -0,0 +1,41 @@
+import isValidDateAfter from './isValidDateAfter';
+import sendSlackMessage from './sendSlackMessgae';
+import Logger from './log';
+
+const log = new Logger('filterFlags');
+
+export default function filterFlags(scheduledFlags, slack) {
+  let description = '';
+  // get only flags that can be deployed
+  return scheduledFlags
+    .reduce((accumulator, f) => {
+      try {
+        description = JSON.parse(f.description);
+      } catch (e) {
+        const message = `${f.key} is scheduled, but its description field is not a valid json object: ${f.description}`;
+        return sendSlackMessage(message, slack);
+      }
+
+      return Array.isArray(description) ? [
+        ...accumulator,
+        ...description.map(value => {
+          if (!isValidDateAfter(value)) return null;
```

Comment:
`isValidDateAfter` should be called `targetDeploymentDatePassed`should not contain isDeployed logic.

This line should be like so:
```
if (!targetDeploymentDatePassed(outstandingTask) || outstandingTask.isDeployed) return null;
```
So the logic is clear.